base/linalg/uniformscaling.jl

```diff
@@ -313,3 +313,27 @@ function hvcat(rows::Tuple{Vararg{Int}}, A::Union{AbstractVecOrMat,UniformScalin
     end
     return hvcat(rows, promote_to_arrays(n,1, promote_to_array_type(A), A...)...)
 end
+
+
+## Cholesky
+function _chol!(J::UniformScaling, uplo)
+    c, info = _chol!(J.λ, uplo)
+    UniformScaling(c), info
+end
+
+chol!(J::UniformScaling, uplo) = ((J, info) = _chol!(J, uplo); @assertposdef J info)
```

Comment:
What is the motivation for this method? Should fall in the same category as the now deprecated `chol(::Number)` IIUC.

Reply:
The function `chol!` is a *chainable modifying array function* and should do the right thing to immutable arrays. If `!`-versions of chainable array functions are defined for immutable arrays, then  `f!(g!(h!(copy(A()))))` is a good way to write efficient and generic code which works for immutables and mutables (here `!` allows `f`, `g`, `h` to use the copy of `A` as workspace where appropriate). In the same fashion, `copy` pushes through immutables and is defined for `UniformScaling`s.